### DISCUSSION TITLE
What are some best practices when capturing data across multiple domains?

### DISCUSSION THREAD

**Discussion by ecormaksin** [spring-projects/spring-modulith#693] [Q&A]
Taking the sample project as an example, when retrieving order information, the customer name would also be needed. In that case, we need to retrieve data across two domains, "order" and "customer. What would be the best practice when retrieving data across multiple domains, and I can't just deal with it in a SQL query, can I?

**Comment by breun** [comment]
Code from one module can call public API of another module. You'd have a `OrderService` to deal with orders and/or a `CustomerService` to deal with customer data. Don't touch another module's database tables directly, that circumvents all of your nice modular architecture.
